sklearn/naive_bayes.py

```diff
@@ -987,3 +1016,206 @@ def _joint_log_likelihood(self, X):
+            It is further assumed that all categories of each feature are
+            represented by the numbers 0, ..., n - 1, where n refers to the
+            total number of categories for the given feature. This can, for
+            instance, be achieved with the help of OrdinalEncoder.
+
+        y : array-like, shape = [n_samples]
+            Target values.
+
+        classes : array-like, shape = [n_classes] (default=None)
+            List of all the classes that can possibly appear in the y vector.
+
+            Must be provided at the first call to partial_fit, can be omitted
+            in subsequent calls.
+
+        sample_weight : array-like, shape = [n_samples], (default=None)
+            Weights applied to individual samples (1. for unweighted).
+
+        Returns
+        -------
+        self : object
+        """
+        return super().partial_fit(X, y, classes,
+                                   sample_weight=sample_weight)
+
+    def _check_X(self, X):
+        # FIXME: we can avoid calling check_array twice after #14872 is merged.
+        # return check_array(X, y, dtype='int', accept_sparse=False,
+        #                   force_all_finite=True)
+        X = check_array(X, accept_sparse=False, force_all_finite=True)
+        return check_array(X, dtype='int')
+
+    def _check_X_y(self, X, y):
+        # FIXME: we can avoid calling check_array twice after #14872 is merged.
+        # return check_array(X, y, dtype='int', accept_sparse=False,
+        #                   force_all_finite=True)
+        X, y = check_X_y(X, y, accept_sparse=False, force_all_finite=True)
+        return check_X_y(X, y, dtype='int')
+
+    def _init_counters(self, n_effective_classes, n_features):
+        self.class_count_ = np.zeros(n_effective_classes, dtype=np.float64)
+        self.category_count_ = [np.zeros((n_effective_classes, 0))
+                                for _ in range(n_features)]
+
+    def _count(self, X, Y):
+        def _update_cat_count_dims(cat_count, diff):
+            if diff > 0:
+                # we append a column full of zeros for each new category
+                return np.pad(cat_count, [(0, 0), (0, diff)], 'constant')
+            return cat_count
+
+        def _update_cat_count(X_feature, Y, cat_count, n_classes):
+            for j in range(n_classes):
+                mask = Y[:, j].astype(bool)
+                if Y.dtype.type == np.int64:
+                    weights = None
+                else:
+                    weights = Y[mask, j]
+                counts = np.bincount(X_feature[mask], weights=weights)
+                indices = np.nonzero(counts)[0]
+                cat_count[j, indices] += counts[indices]
+
+        self.class_count_ += Y.sum(axis=0)
+        for i in range(self.n_features_):
+            X_feature = X[:, i]
+            diff = X_feature.max() + 1 - self.category_count_[i].shape[1]
+            self.category_count_[i] = _update_cat_count_dims(
```

Comment:
I think `self.category_count_[i] = _update_cat_count_dims(self.category_count_[i], X_feature.max())` would be a more intelligible interface than defining `diff` here.